web3-dev/smart-contracts/smart-contract-example-sum.rst

```diff
@@ -124,61 +84,63 @@ Smart-contract can be compiled using the command:
 
     npm run build
 
+Note that a `build/deployer.wasm` file has also been generated. It will be used to deploy your contract on Massa network.
+
 .. _sending-sc-sum:
 
-Putting your smart-contract on the blockchain
----------------------------------------------
+Deploy your smart-contract on the blockchain
+--------------------------------------------
 
 We'll now turn to the process of putting the smart-contract on the Massa blockchain.
 
 For the deployment, you will need a wallet with some coins. 
 
+To send transaction on the network, you need to provide your wallet private key.
+This is done using environment variable in `.env` file.
+
 .. code-block::
 
     cp .env.example .env
 
-This command will create a `.env` file. Now fill it with your wallet keys and address.
+This command will create a `.env` file. Now fill it with your wallet private key.
 
-Deploying a smart-contract on Massa blockchain is done by calling a temporary smart-contract that will store 
-our sum smart-contract onto the ledger.
+Contract deployment is done by calling a deployer smart-contract which can be found here: `deployer/deployer.as.ts`. It will store our sum smart-contract onto the ledger. The deployer contract already includes your compiled `main.ts` contract and has been itself compiled at the `npm run build` step.
 
-Sending the smart-contract to the Massa blockchain is done with the command `npm run deploy`.
+We will send the deployer smart-contract to the Massa blockchain with:
 
 .. code-block::
 
-    npm run deploy build/main.wasm
+    npm run deploy
 
-This command will execute `main.ts` smart-contract, and this smart-contract will store the `sum.ts` smart-contract onto the ledger.
+This command will execute the compiled deployer `deployer.wasm`, and this smart-contract will store the `main.ts` smart-contract onto the ledger.
 
 You will see an output like this:
 
 .. code-block::
 
-    > my-massa-sc@1.0.0 deploy
-    > ts-node --esm deployer/deployment_script.ts build/main.wasm
+    > my-massa-sc@0.0.1 deploy
```

Comment:
can't we use latest ? Or do you think it's too dangerous ?

Reply:
I update to     > my-massa-sc@0.0.1 deploy

Reply:
What about sc-example-sum@latest ?

Reply:
this is an example of the output of the command I don't know if we want to control it